### DISCUSSION TITLE
Dynamically adding wizard steps

### DISCUSSION THREAD

**Discussion by kolaente** [filamentphp/filament#12463] [Help]
### Package

Form builder

### Package Version

3.2.71

### How can we help you?

I'm trying to add a wizard step based on the select value of a radio button of a previous step. I notices the `Step` component has both a `live()` and a `hidden()` value, but it does not seem to work as expected.

Here's what I tried:

```php
    protected function getSteps(): array
    {
        return [
            Step::make(__('Type'))
                ->live()
                ->schema([
                    Forms\Components\Radio::make('type')
                        ->label(__('Type'))
                        ->live()
                        ->required()
                        ->options(TypeEnum::class)
                ]),
            Step::make(__('Appearance'))
                ->hidden(fn(Forms\Get $get): bool => $get('type') !== TypeEnum::BUBBLE->value)
                ->schema([
                     // … 
                ]),
        ];
    }
```

**Comment by leandrocfe** [comment]
Are you using the Wizard in a Page Resource?

Does this example work?

```php
Step::make(__('Type'))
    ->live()
    ->schema([
        Radio::make('type')
            ->label(__('Type'))
            ->live()
            ->required()
            ->options(
                [
                    'type1' => 'Type 1',
                    'type2' => 'Type 2',
                    'type3' => 'Type 3',
                ]
            ),
    ]),
Step::make(__('Appearance'))
    ->hidden(fn (Get $get): bool => $get('type') !== 'type1')
    ->schema([]),
```

**Reply by kolaente** [reply]
Yes, I am using it in a Page Resource. I've tried what you suggested, without luck (similar to the code I shared initially)

**Reply by leandrocfe** [reply]
Could you share the whole code?

**Reply by kolaente** [reply]
Trying to reproduce it, I discovered the `Get` function always seems to return a string. This means this will not work because of the typed comparison:

```php
Step::make(__('Type'))
    ->live()
    ->schema([
        Forms\Components\Radio::make('type')
            ->label(__('Type'))
            ->live()
            ->required()
            ->options(
                [
                    1 => 'Type 1',
                    2 => 'Type 2',
                    3 => 'Type 3',
                ]
            ),
    ]),
Step::make(__('Appearance'))
    ->hidden(fn (Get $get): bool => $get('type') !== 1)
    ->schema([]),
```

If I change the comparison to `->hidden(fn (Get $get): bool => $get('type') != 1)` (weak) it works.

In my code (see the first example), I'm using an `int` based enum so I guess Filament will create an option field with ints as the value, similar to the code example above.

To me, this looks like a bug - the `Get` method should always return the original value, not cast it to a string. Should I open a bug report?

**Reply by kolaente** [reply]
Another observation in a similar direction: The comparison works with `!==` when I compare on fields on the edit page, but not when the same field is part of a wizard step.

This kind of field definition:

```php
Forms\Components\TextInput::make('headline')
    ->hidden(fn (Forms\Get $get): bool => $get('type') !== TypeEnum::BUBBLE->value)
    ->required()
```
